src/frontend/src/handler/create_source.rs

```diff
@@ -798,7 +798,10 @@ pub(crate) async fn bind_source_pk(
 
         // For all Upsert formats, we only accept one and only key column as primary key.
         // Additional KEY columns must be set in this case and must be primary key.
-        (Format::Upsert, encode @ Encode::Json | encode @ Encode::Avro) => {
+        (
+            Format::Upsert,
+            encode @ Encode::Json | encode @ Encode::Avro | encode @ Encode::Protobuf,
```

Comment:
Given the comment above says *all Upsert formats*, shall we match `(Format::Upsert, _)` ?

Reply:
It makes sense for all encodes other than `Encode::Native` but we may proceed with caution. Let's add more encodes here as requested.